dashboard/src/shared/Auth.ts

```diff
@@ -77,4 +80,18 @@ export class Auth {
     }
     return null;
   }
+
+  // defaultNamespaceFromToken decodes a jwt token to return the k8s service
+  // account namespace.
+  // TODO(mnelson): until we call jwt.verify on the token during validateToken above
+  // we use a default namespace for both invalid tokens and tokens without the expected
+  // key.
+  public static defaultNamespaceFromToken(token: string) {
+    const payload = jwt.decode(token);
+    const namespaceKey = "kubernetes.io/serviceaccount/namespace";
+    if (!payload || !payload[namespaceKey]) {
+      return DEFAULT_NAMESPACE;
+    }
+    return payload[namespaceKey] || DEFAULT_NAMESPACE;
```

Comment:
You're checking it `payload[namespaceKey]` is defined in the previous conditional. I'd remove the `|| DEFAULT_NAMESPACE` clause from here to avoid logic duplication.

Reply:
Woops, left after a refactor. Thanks.